evalml/pipelines/component_graph.py

```diff
@@ -271,7 +286,7 @@ def _compute_features(self, component_list, X, y=None, fit=False):
                     output = component_instance.predict(input_x)
                 else:
                     output = None
-                output_cache[component_name] = output
+                output_cache[f"{component_name}.x"] = output
```

Comment:
This means that components will use estimator outputs as `estimator_name.x`, not `estimator_name` to align with how users need to specify :)